### PR TITLE
Re-write edit client relationship management form in React

### PR DESCRIPTION
**Pull request by mforner13**
## Description of change

This PR re-writes the form to edit an investment project's client relationship manager in react.

This is due to the previous implementation using the vue.js typeahead, and part of a wider task to remove vue.js from the codebase.

Includes:
- Removing old logic from express
- Removing unit test for form/controller
- Adding a functional test spec for the form
- Moving react project team code to the `investments/client/projects/team` folder
- Adding client-side validation that prevents you removing a client relationship manager without adding a new one (previously a user was able to remove it and save the form, however this is a mandatory field in the API for a project so the change was not actually made)

## Test instructions

Go to an investment project
- Go to the project team tab
- Edit the client relationship manager
- You should be able to update the CRM and save the form
- You should not be able to save the form if there is no CRM

### Before

Screenshot 2022-02-28 at 14 29 31

### After

Screenshot 2022-02-28 at 14 29 17

## Checklist

[//]: # "When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/master/docs/Code%20review%20guidelines.md"

- [ ] Has the branch been rebased to master?
- [ ] Automated tests (Any of the following when applicable: Unit, Functional or End-to-End)
- [ ] Manual compatibility testing (Browsers: Chrome, Firefox, Edge, Safari)

### CODE REVIEW

File: src/apps/investments/client/projects/team/GlobalAccountManagerDetails.jsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    </Paragraph>
+    <Details
+      summary="Need to edit the Global Account Manager information?"
+      data-test="global-account-manager-links"
+    >
+      If you need to change the Global Account Manager for this company, go to
+      the{' '}
+      <NewWindowLink
+        showHelpText={false}
+        href="https://workspace.trade.gov.uk/working-at-dit/policies-and-guidance/strategic-relationship-account-management/"
```
Comment (cgsunkel) [inline src/apps/investments/client/projects/team/GlobalAccountManagerDetails.jsx:31]: Can this be moved into the URLs module?

File: src/apps/investments/client/projects/team/GlobalAccountManagerDetails.jsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        ariaLabel="Digital Workspace"
+      >
+        Digital Workspace
+      </NewWindowLink>
+      {' or '}
+      {oneListEmail && (
+        <>
+          <NewWindowLink
```
Comment (cgsunkel) [inline src/apps/investments/client/projects/team/GlobalAccountManagerDetails.jsx:39]: I'm not sure `NewWindowLink` is appropriate here, there aren't any other `mailto` links using it

File: src/client/components/NewWindowLink/index.jsx [hunk 3]
```diff
@@ -2,29 +2,41 @@ import React from 'react'
 import Link from '@govuk-react/link'
 import PropTypes from 'prop-types'
 
-const NewWindowLink = ({ href, showHelpText, children, ...rest }) => (
+const DEFAULT_HELPTEXT = 'Opens in a new window or tab'
```
Comment (cgsunkel) [inline src/client/components/NewWindowLink/index.jsx:5]: Can you update the `NewWindowLink` component test to include these changes?

File: test/cypress/plugins/index.js [hunk 4]
```diff
@@ -3,17 +3,21 @@ require('dotenv').config()
 
 module.exports = (on, config) => {
   if (config.testingType === 'component') {
-    require('@cypress/react/plugins/load-webpack')(on, config, {webpackFilename: './webpack.config.js'})
+    require('@cypress/react/plugins/load-webpack')(on, config, { webpackFilename: './webpack.config.js' })
```
Comment (cgsunkel) [inline test/cypress/plugins/index.js:6]: Were the changes to this file intentional?

File: src/apps/investments/client/projects/team/GlobalAccountManagerDetails.jsx (hunk 2 above)
Comment (cgsunkel) [reply src/apps/investments/client/projects/team/GlobalAccountManagerDetails.jsx:39]: We could also move all the `mailto` links into the URLs module as well so we don't have any hardcoded email addresses floating around (I might have some capacity to sort out the existing ones)

Comment (mforner13) [reply src/apps/investments/client/projects/team/GlobalAccountManagerDetails.jsx:40]: Agreed it doesn't need to be a new window link as mailtos open in a new window anyway :)

This email address comes from config - `oneListEmail: config.oneList.email,` this is passed from the controller, so isn't hardcoded

File: src/client/components/NewWindowLink/index.jsx (hunk 3 above)
Comment (mforner13) [reply src/client/components/NewWindowLink/index.jsx]: I've reverted the changes as were no longer needed now I switched the mailto link to use the standard 'Link' :)

**Review by cgsunkel — APPROVED** [review]

**Pull request merged**